File: app/src/main/java/co/tpcreative/supersafe/common/views/SquaredView.java
package co.tpcreative.supersafe.common.views;
import android.content.Context;
import android.util.AttributeSet;
import android.view.View;


public class SquaredView extends View {
    public SquaredView(Context context) {
        super(context);
    }

    public SquaredView(Context context, AttributeSet attrs) {
        super(context, attrs);
    }

    @Override
    protected void onMeasure(int widthMeasureSpec, int heightMeasureSpec) {
        super.onMeasure(widthMeasureSpec, heightMeasureSpec);
        setMeasuredDimension(getMeasuredWidth(), getMeasuredWidth());
    }

    public void toggle(){
        setVisibility(getVisibility()==VISIBLE?INVISIBLE:VISIBLE);
    }
}

File: app/src/main/java/co/tpcreative/supersafe/common/views/SquaredView.kt
package co.tpcreative.supersafe.common.views
import android.content.Context
import android.util.AttributeSet
import android.view.View

class SquaredView : View {
    constructor(context: Context?) : super(context) {}
    constructor(context: Context?, attrs: AttributeSet?) : super(context, attrs) {}

    override fun onMeasure(widthMeasureSpec: Int, heightMeasureSpec: Int) {
        super.onMeasure(widthMeasureSpec, heightMeasureSpec)
        setMeasuredDimension(measuredWidth, measuredWidth)
    }

    fun toggle() {
        visibility = if (visibility == VISIBLE) INVISIBLE else VISIBLE
    }
}